### PR TITLE
Laser pipes

### PR DESCRIPTION
**Pull request by brachy84**
This is only a basic implementation. Rendering still doesn't work for me.
Feel free to change anything to your like.

EDIT: I managed to fix item rendering. The block is still invisible

### CODE REVIEW

**Comment by htmlcsjs** [comment]
Thank you for the work, Ill try to see what's wrong with the rendering, but Im not the best with rendering tbh

File: src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        if (tileEntityPipe == null) {
+            GTLog.logger.info("Tile is null");
+            return false;
+        }
+
+        LaserPipeType fluidPipeType = tileEntityPipe.getPipeType();
+        int paintingColor = tileEntityPipe.getInsulationColor();
+        int connectedSidesMap = blockFluidPipe.getVisualConnections(tileEntityPipe);
+
+        if (fluidPipeType != null) {
```
Comment (brachy84) [inline src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java:126]: This must be the reason the block doesn't render, but I don't know why it is null

File: src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            GTLog.logger.info("Rendering laser pipe block");
```
Comment (htmlcsjs) [inline src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java:127]: Can this please be changed to `htmlTech.logger`

File: src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+public class LaserPipeRenderer implements ICCBlockRenderer, IItemRenderer {
+
+    public static final ModelResourceLocation MODEL_LOCATION = new ModelResourceLocation(new ResourceLocation(GTValues.MODID, "laser_pipe_normal"), "normal");
+    public static final LaserPipeRenderer INSTANCE = new LaserPipeRenderer();
+    public static EnumBlockRenderType BLOCK_RENDER_TYPE;
+    private static final ThreadLocal<BlockRenderer.BlockFace> blockFaces = ThreadLocal.withInitial(BlockRenderer.BlockFace::new);
+    private final Map<LaserPipeType, PipeTextureInfo> pipeTextures = new HashMap();
+
+    public static void preInit() {
+        GTLog.logger.info("initializing laser pipe renderer");
```
Comment (htmlcsjs) [inline src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java:62]: Can this please be changed to `htmlTech.logger`

File: src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+            TextureAtlasSprite inTexture = map.registerSprite(inLocation);
+            TextureAtlasSprite sideTexture = map.registerSprite(sideLocation);
+            this.pipeTextures.put(laserPipeType, new PipeTextureInfo(inTexture, sideTexture));
+        }
+    }
+
+    @SubscribeEvent
+    public void onModelsBake(ModelBakeEvent event) {
+        GTLog.logger.info("registering laser pipe model");
```
Comment (htmlcsjs) [inline src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java:82]: Can this please be changed to `htmlTech.logger`

File: src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        renderState.reset();
+        renderState.bind(buffer);
+        renderState.setBrightness(world, pos);
+        IVertexOperation[] pipeline = {new Translation(pos)};
+
+        BlockLaserPipe blockFluidPipe = ((BlockLaserPipe) state.getBlock());
+        TileEntityLaserPipe tileEntityPipe = (TileEntityLaserPipe) blockFluidPipe.getPipeTileEntity(world, pos);
+
+        if (tileEntityPipe == null) {
+            GTLog.logger.info("Tile is null");
```
Comment (htmlcsjs) [inline src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java:118]: Can this please be changed to `htmlTech.logger`

**Review by htmlcsjs — COMMENTED** [review]
This is just something i noticed when trying to debug the pipe rendering

**Comment by htmlcsjs** [comment]
How ready is this for merge?

**Pull request merged**